### PR TITLE
chore: encapsulate all Database calls through State

### PR DESCRIPTION
**Pull request by sborrazas**
This also includes retrieving the State from the State plug in all
controllers which were missing it.

### CODE REVIEW

**Review by thepiwo — APPROVED** [review]

**Review by jyeshe — CHANGES_REQUESTED** [review]

File: lib/ae_mdw/db/contract.ex [hunk 1]
```diff
@@ -354,13 +315,13 @@ defmodule AeMdw.Db.Contract do
...
-  @spec aex9_search_contract(pubkey(), integer()) :: map()
-  def aex9_search_contract(account_pk, last_txi) do
+  @spec aex9_search_contract(State.t(), pubkey(), integer()) :: map()
+  def aex9_search_contract(state, account_pk, last_txi) do
     gen_collect(
       Model.Aex9AccountPresence,
       {account_pk, last_txi, <<max_256bit_int()::256>>},
       fn {acc_pk, _, _} -> acc_pk == account_pk end,
-      &prev/2,
+      &State.prev(state, &1, &2),
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply lib/ae_mdw/db/contract.ex]: Good call, done

**Review by thepiwo — APPROVED** [review]

File: lib/ae_mdw/db/contract.ex [hunk 2]
```diff
@@ -317,19 +316,10 @@ defmodule AeMdw.Db.Contract do
...
-        Map.update(accum, ct_pk, [txi], fn txi_list ->
-          [txi | txi_list]
-        end)
-      end,
-      & &1
-    )
+    state
+    |> Collection.stream(Model.Aex9AccountPresence, {account_pk, last_txi, Util.min_int()})
+    |> Stream.take_while(&match?({^account_pk, _txi, _contract_pk}, &1))
+    |> Enum.group_by(fn {_account_pk, _txi, contract_pk} -> contract_pk end)
```
Comment (jyeshe) [inline lib/ae_mdw/db/contract.ex:322]: need to return a mapping from `ct_pk` to `txi_list`

Comment (sborrazas) [reply lib/ae_mdw/db/contract.ex:322]: That's what [`Enum.group_by/2`](https://hexdocs.pm/elixir/1.13/Enum.html#group_by/3) does

Comment (sborrazas) [reply lib/ae_mdw/db/contract.ex]: nvm! I just realized that the values of the map weren't a txi list, my bad! fixed it now

**Review by jyeshe — APPROVED** [review]

**Pull request merged**